junto-rust/zomes/core/code/src/expressions/channel.rs

```diff
@@ -0,0 +1,123 @@
+//Module to handle all channel related operations
+use hdk::{
+    error::ZomeApiResult,
+    error::ZomeApiError,
+    holochain_core_types::{
+        entry::Entry, 
+        cas::content::Address
+    }
+};
+
+use super::utils;
+use super::group;
+use super::user;
+use super::definitions::{
+    app_definitions,
+    function_definitions::{
+        FunctionDescriptor,
+        FunctionParameters
+    }
+};
+
+//Commits den entry to DHT and runs necassary hooks
+pub fn commit_den(entry: &Entry, user: &Address) -> Result<Address, String> {
+    let pack_address;
+    match hdk::commit_entry(&entry){
+        Ok(address) => {
+            pack_address = address.clone();
+            //Build vector describing hook functions which should run to correctly link this data
+            let hook_definitions = vec![FunctionDescriptor{name: "link_expression", parameters: FunctionParameters::LinkExpression{tag: "den", direction: "reverse", parent_expression: address.clone(), child_expression: user.clone()}},
+                                        FunctionDescriptor{name: "link_expression", parameters: FunctionParameters::LinkExpression{tag: "owner", direction: "forward", parent_expression: address.clone(), child_expression: user.clone()}}];
+
+            match utils::handle_hooks("Channel".to_string(), hook_definitions){
+                Ok(_result) => {},
+                Err(hdk_err) => return Err(hdk_err.into()),
+            }
+        },
+        Err(_hdk_err) => return Err("There was an error commiting den entry".to_string())
+    }
+    Ok(pack_address)
+}
+
+//Create den(s) (multiple dens as signified by app_definitions data) and link to user with required tags as defined by definitons data
+pub fn create_den(username_address: &Address, first_name: String) -> ZomeApiResult<serde_json::Value> {
```

Comment:
We suggest using strong types and avoiding the json! macro where possible. So maybe you could make a CreateDenReturnValue struct to return so more checking happens at compile time. This is a matter of preference though